retrieval/manager.go

```diff
@@ -103,8 +107,25 @@ func (r *PrometheusReader) CurrentSegment() int {
 	return r.tailer.CurrentSegment()
 }
 
+// getjobInstanceMap returns a string map for any job for which the instance
+// label has been relabeled
+func (r *PrometheusReader) getJobInstanceMap() map[string]string {
+	jobInstanceMap := make(map[string]string)
+	for _, config := range r.scrapeConfig {
+		for _, metricRelabel := range config.MetricRelabelConfigs {
+			for _, ln := range metricRelabel.SourceLabels {
+				if string(ln) == "instance" {
```

Comment:
We should test `metricRelabel.Action` is defaulted ("" or "replace"), to be conservative. It seems this works even when `len(metricRelabel.SourceLabels) != 1`.

(The default action is "replace" according to https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config, but you could see "instance" in SourceLabels for other rules, where we probably don't need to update `jobInstanceMap`.)

Something like `if metricRelabel.Action != "" && metricRelabel.Action != "replace" { continue }`.

I wonder if we will see cases where `config.RelabelConfigs` section (as opposed to `config.MetricRelabelConfigs`) has to be scanned, but don't want to complicate this PR. Also I wonder if we will see more-bizarre multi-step relabelings that break this logic. In the future, I imagine, we could add a new section to the sidecar's config file with an explicit mapping, in which case users could bypass this implicit logic.

Reply:
Looking at the documentation you linked, I wonder if the right thing to test for is a `labeldrop` for the `instance` label instead, since a replace doesnt necessarily mean the instance label will be removed.

Reply:
I _think_ a label that includes the `instance` concatenated with other label values will also work. If there's a labeldrop _after_ the replace, then we'll have needed to recognize the replace, but you're right that if there's been a replace we don't have to pay attention unless there's also a labeldrop.

To me, the confusion is that Prometheus "replace" does not actually drop the labels, it just outputs a new label. Misleading, right?

Reply:
Right, replace means take the value and replace it with whatever rule is configured into a new target label :D